File: src/javascript/user/contact/list-contacts.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const client = new FaxCoreClient();

const result = await client.post('/api/contact/list', {
  addressBookID: Number(process.env.FAXCORE_ADDRESS_BOOK_ID || 0),
  isContactGroup: false,
  addressType: ['Fax', 'Email'],
  pagination: {
    search: process.env.FAXCORE_CONTACT_SEARCH || '',
    page: Number(process.env.FAXCORE_PAGE || 1),
    maxResult: Number(process.env.FAXCORE_MAX_RESULT || 25)
  }
});

printResult('Contact list', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Contact/ListContacts.cs
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Contact
{
    internal static class ListContacts
    {
        public static Task<string> RunAsync(FaxCoreClient client)
        {
            var request = new
            {
                addressBookID = 0,
                isContactGroup = false,
                addressType = new[] { "Fax", "Email" },
                pagination = new
                {
                    search = string.Empty,
                    page = 1,
                    maxResult = 25
                }
            };

            return client.PostAsync("/api/contact/list", request);
        }
    }
}

